pennylane/qnode_new/jacobian.py

```diff
@@ -0,0 +1,528 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Differentiable quantum nodes.
+"""
+
+from collections.abc import Sequence
+import copy
+
+import autograd.extend
+import autograd.builtins
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import ObservableReturnTypes, CV
+from pennylane.utils import _flatten, unflatten, _inv_dict
+from pennylane.qnode_new.qnode import QNode, QuantumFunctionError
+
+
+
+class JacobianQNode(QNode):
+    """Quantum node that can be differentiated with respect to its positional parameters.
+    """
+    interface = None  #: str, None: automatic differentiation interface used by the node, if any
+
+    def __init__(self, func, device, mutable=True, properties=None):
+        super().__init__(func, device, mutable=mutable, properties=properties)
+
+        #: dict[int, str]: map from free parameter index to the gradient method to be used with that parameter
+        self.par_to_grad_method = None
+
+    def __repr__(self):
+        """String representation."""
+        detail = "<JNode: device='{}', func={}, wires={}, interface={}>"
+        return detail.format(self.device.short_name, self.func.__name__, self.num_wires, self.interface)
+
+    def _construct(self, args, kwargs):
+        """Constructs the quantum circuit graph by calling the quantum function.
+
+        Like :meth:`.QNode._construct`, additionally determines the best gradient computation method
+        for each free parameter.
+        """
+        super()._construct(args, kwargs)
+        self.par_to_grad_method = {k: self._best_method(k) for k in self.variable_deps}
+
+        temp = [str(ob) for ob in self.circuit.observables if ob.return_type is ObservableReturnTypes.Sample]
```

Comment:
why is `str` needed here?

Reply:
`join` works on strings, but `join` is only required for cosmetic purposes. Now the output is `op1; op2; op3`, if we just used "{}".format(list_of_ops) it'd be `[op1, op2, op3]`